src/content/polyfill.ts

```diff
@@ -0,0 +1,122 @@
+        this.addEventListener('monetization', val)
+        handlers.set(this, val)
+      } else {
+        throw new Error('val must be a function, got ' + typeof val)
+      }
+    }
+  }
+
+  const supportsOriginal = DOMTokenList.prototype.supports
+  const supportsMonetization = Symbol.for('link-supports-monetization')
+  DOMTokenList.prototype.supports = function (token) {
+    // @ts-expect-error: polyfilled
+    if (this[supportsMonetization] && token === 'monetization') {
+      return true
+    } else {
+      return supportsOriginal.call(this, token)
+    }
+  }
+
+  const relList = Object.getOwnPropertyDescriptor(
+    HTMLLinkElement.prototype,
+    'relList'
+  )!
+  const relListGetOriginal = relList.get!
+
+  relList.get = function () {
+    const val = relListGetOriginal.call(this)
+    val[supportsMonetization] = true
+    return val
+  }
+
+  Object.defineProperty(HTMLLinkElement.prototype, 'relList', relList)
+  Object.defineProperty(HTMLElement.prototype, 'onmonetization', attributes)
+  Object.defineProperty(Window.prototype, 'onmonetization', attributes)
+  Object.defineProperty(Document.prototype, 'onmonetization', attributes)
+
+  let eventDetailDeprecationEmitted = false
+  class MonetizationEvent extends Event {
+    public readonly amountSent: PaymentCurrencyAmount
+    public readonly incomingPayment: string
+    public readonly paymentPointer: string
+
+    constructor(details: MonetizationEventPayload['detail']) {
+      super('monetization', { bubbles: true })
+      const { amountSent, incomingPayment, paymentPointer } = details
+      this.amountSent = amountSent
+      this.incomingPayment = incomingPayment
+      this.paymentPointer = paymentPointer
+    }
+
+    get [Symbol.toStringTag]() {
+      return 'MonetizationEvent'
+    }
+
+    get detail() {
+      if (!eventDetailDeprecationEmitted) {
+        const msg = `MonetizationEvent.detail is deprecated. Access attributes directly instead.`
+        // eslint-disable-next-line no-console
+        console.warn(msg)
+        eventDetailDeprecationEmitted = true
+      }
+      const { amountSent, incomingPayment, paymentPointer } = this
+      return { amountSent, incomingPayment, paymentPointer }
+    }
+  }
+
+  // @ts-expect-error: we're defining this now
+  window.MonetizationEvent = MonetizationEvent
+
+  window.addEventListener(
+    '__wm_ext_monetization',
+    (event: CustomEvent<MonetizationEventPayload['detail']>) => {
+      if (!(event.target instanceof HTMLLinkElement)) return
+      if (!event.target.isConnected) return
+
+      const monetizationTag = event.target
+      monetizationTag.dispatchEvent(new MonetizationEvent(event.detail))
+    },
+    { capture: true }
+  )
+
+  window.addEventListener(
+    '__wm_ext_onmonetization_attr_change',
+    (event: CustomEvent<{ attribute?: string }>) => {
+      const link = event.target
+      if (!(link instanceof HTMLLinkElement)) return
+      if (!link.isConnected) return
```

Comment:
The `onmonetization` event handler can be present on any element:

> The onmonetization event handler is an[ event handler content attribute](https://html.spec.whatwg.org/multipage/webappapis.html#event-handler-content-attributes) that can be applied to any [element](https://dom.spec.whatwg.org/#concept-element). The user agent uses it to notify that some link has been [monetized](https://webmonetization.org/specification/#dfn-monetized).

Reply:
Oh. Fixing..

Reply:
btw, monetization event is on Line 95 above.